zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,870 @@
+    resources
+        The explicitly passed resources to compute expr.
+
+    Raises
+    ------
+    ValueError
+        If the resources to not match for an expression.
+    """
+    if expr is None:
+        return
+    bound = expr._resources()
+    if not bound and resources is None:
+        raise ValueError('no resources provided to compute %s' % name)
+    if bound and resources:
+        raise ValueError(
+            'explicit and implicit resources provided to compute %s' % name,
+        )
+
+
+def _check_datetime_field(name, measure):
+    """Check that a field is a datetime inside some measure.
+
+    Parameters
+    ----------
+    name : str
+        The name of the field to check.
+    measure : Record
+        The record to check the field of.
+
+    Raises
+    ------
+    TypeError
+        If the field is not a datetime inside ``measure``.
+    """
+    if not isinstance(measure[name], (Date, DateTime)):
+        raise TypeError(
+            "'{name}' field must be a '{dt}', not: '{dshape}'".format(
+                name=name,
+                dt=DateTime(),
+                dshape=measure[name],
+            ),
+        )
+
+
+class NoDeltasWarning(UserWarning):
+    """Warning used to signal that no deltas could be found and none
+    were provided.
+
+    Parameters
+    ----------
+    expr : Expr
+        The expression that was searched.
+    """
+    def __init__(self, expr):
+        self._expr = expr
+
+    def __str__(self):
+        return 'No deltas could be infered from expr: %s' % self._expr
+
+
+_valid_no_deltas_rules = 'warn', 'raise', 'ignore'
+
+
+def _get_deltas(expr, deltas, no_deltas_rule):
+    """Find the correct deltas for the expression.
+
+    Parameters
+    ----------
+    expr : Expr
+        The base expression.
+    deltas : Expr, 'auto', or None
+        The deltas argument. If this is 'auto', then the deltas table will
+        be searched for by walking up the expression tree. If this can not be
+        reflected, then an action will be taken based on the 'no_deltas_rule'.
+    no_deltas_rule : {'warn', 'raise', 'ignore'}
+        How to handle the case where deltas='auto' but no deltas could be
+        found.
+
+    Returns
+    -------
+    deltas : Expr or None
+        The deltas table to use.
+    """
+    if no_deltas_rule not in _valid_no_deltas_rules:
+        raise ValueError(
+            'no_deltas_rule must be one of: %s' %
+            _valid_no_deltas_rules
+        )
+
+    if isinstance(deltas, bz.Expr) or deltas != 'auto':
+        return deltas
+
+    try:
+        return expr._child[(expr._name or '') + '_deltas']
+    except (ValueError, AttributeError):
+        if no_deltas_rule == 'raise':
+            raise ValueError(
+                "no deltas table could be reflected for %s" % expr
+            )
+        elif no_deltas_rule == 'warn':
+            warnings.warn(NoDeltasWarning(expr))
+    return None
+
+
+def from_blaze(expr,
```

Comment:
There are a lot of individually-reasonable blocks in this function that add up to make it pretty hard to reason about all at once.  Could we break this up into more atomic checks?